testing/test-setup-config/src/lib/vitest-setup-presets.unit.test.ts

```diff
@@ -0,0 +1,331 @@
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+import type { VitestOverrides } from './vitest-config-factory.js';
+import * as configFactory from './vitest-config-factory.js';
+import {
+  createE2eConfig,
+  createIntConfig,
+  createUnitConfig,
+  setupPresets,
+} from './vitest-setup-presets.js';
+
+vi.mock('./vitest-config-factory.js', () => ({
+  createVitestConfig: vi.fn().mockReturnValue('mocked-config'),
+}));
+
+const MOCK_PROJECT_KEY = 'test-package';
+const MOCK_CONFIG_REST_PARAMS = {
+  projectRoot: '/test/project',
+  cacheKey: 'test-cache',
+};
+
+const TEST_TIMEOUTS = {
+  SHORT: 5000,
+  MEDIUM: 10_000,
+  LONG: 15_000,
+  E2E: 60_000,
+} as const;
+
+describe('setupPresets', () => {
+  it('should export correct unit setup presets', () => {
+    expect(setupPresets.unit).toEqual({
+      base: [
+        'testing/test-setup/src/lib/console.mock.ts',
+        'testing/test-setup/src/lib/reset.mocks.ts',
+        'testing/test-setup/src/lib/cliui.mock.ts',
+        'testing/test-setup/src/lib/fs.mock.ts',
+        'testing/test-setup/src/lib/extend/ui-logger.matcher.ts',
+      ],
+      git: ['testing/test-setup/src/lib/git.mock.ts'],
+      portalClient: ['testing/test-setup/src/lib/portal-client.mock.ts'],
+      matchersCore: [
+        'testing/test-setup/src/lib/extend/markdown-table.matcher.ts',
+        'testing/test-setup/src/lib/extend/jest-extended.matcher.ts',
+      ],
+      matcherPath: ['testing/test-setup/src/lib/extend/path.matcher.ts'],
+    });
+  });
+
+  it('should export correct integration setup presets', () => {
+    expect(setupPresets.int).toEqual({
+      base: [
+        'testing/test-setup/src/lib/console.mock.ts',
+        'testing/test-setup/src/lib/reset.mocks.ts',
+      ],
+      cliui: ['testing/test-setup/src/lib/cliui.mock.ts'],
+      fs: ['testing/test-setup/src/lib/fs.mock.ts'],
+      git: ['testing/test-setup/src/lib/git.mock.ts'],
+      portalClient: ['testing/test-setup/src/lib/portal-client.mock.ts'],
+      matcherPath: ['testing/test-setup/src/lib/extend/path.matcher.ts'],
+      chromePath: ['testing/test-setup/src/lib/chrome-path.mock.ts'],
+    });
+  });
+
+  it('should export correct e2e setup presets', () => {
+    expect(setupPresets.e2e).toEqual({
+      base: ['testing/test-setup/src/lib/reset.mocks.ts'],
+    });
+  });
+
+  it('should be defined as a const object', () => {
+    expect(setupPresets).toBeDefined();
+    expect(typeof setupPresets).toBe('object');
+  });
+});
+
+// Parameterized tests to eliminate duplication
+describe.each([
+  ['unit', setupPresets.unit.base, createUnitConfig],
+  ['int', setupPresets.int.base, createIntConfig],
+  ['e2e', setupPresets.e2e.base, createE2eConfig],
+] as const)('%s config creation', (kind, baseSetupFiles, createFn) => {
```

Comment:
The each block for describe is introducing conditional checks in the it blocks as well as const. It further makes the tests harder to read. Let's avoid conditionals in tests as good as possible. 

If you want to test basic values you could have a separate it block using the each for static things like defaults that are always the same. In the specific cases you can use `expect.objectContaining` to keep is focused.

se [example for conditionaly](https://github.com/code-pushup/cli/pull/1107/files#r2403979638)